src/boot/efi/devicetree.c

```diff
@@ -104,6 +106,65 @@ EFI_STATUS devicetree_install(struct devicetree_state *state, EFI_FILE *root_dir
                         MAKE_GUID_PTR(EFI_DTB_TABLE), PHYSICAL_ADDRESS_TO_POINTER(state->addr));
 }
 
+static char* devicetree_get_compatible(const void *dtb) {
+        const struct fdt_header *dt_header = dtb;
+        uint32_t *cursor = (uint32_t *) ((char *)dt_header + be32toh(dt_header->off_dt_struct));
```

Comment:
hmm, you area dereferencing an offset here that originates in untrusted data, no? that's no ok, you need to validate this against the size of the dtb object

Reply:
also, alignment is weird here, the offset might not be 32bit aligned yet you just dereference the value

Reply:
Yes, I will add these checks in the updated version